pkg/option/config.go

```diff
@@ -3354,8 +3370,24 @@ func InitConfig(programName, configName string) func() {
 			} else {
 				// replace deprecated fields with new fields
 				ReplaceDeprecatedFields(m)
-				err := MergeConfig(m)
-				if err != nil {
+
+				var confArgs []string
+				var flags = viper.AllKeys()
+				for key, value := range m {
+					val := fmt.Sprintf("%v", value)
+					// check whether the flag exists
+					if IsKnownFlag(flags, key) {
+						if val != "" {
```

Comment:
nit: this check can be move up right after assignment of val e.g.

```go
val := fmt.Sprintf("%v", value)
if val == "" {
    continue
}
```

Reply:
> nit: this check can be move up right after assignment of val e.g.
> 
> ```go
> val := fmt.Sprintf("%v", value)
> if val == "" {
>     continue
> }
> ```

Thanks. Done